src/app/modules/shared_models/collectibles_model.nim

```diff
@@ -262,6 +283,44 @@ QtObject:
     self.endRemoveRows()
     self.countChanged()
 
+  proc updateCollectibleItems(self: Model, newItems: seq[CollectiblesEntry]) =
+    if len(self.items) == 0:
+      self.resetCollectibleItems(newItems)
+      return
+    
+    if len(newItems) == 0:
+      self.removeCollectibleItems()
+      return
+
+    var newTable = initTable[string, int](len(newItems))
+    for i in 0 ..< len(newItems):
+      newTable.add(newItems[i].getID(), i)
+
+    # Needs to be built in sequential index order
+    var oldIndicesToRemove: seq[int] = @[]
+    for idx in 0 ..< len(self.items):
+      let uid = self.items[idx].getID()
+      if not newTable.hasKey(uid):
+        # Item in old list but not in new -> Must remove
+        oldIndicesToRemove.add(idx)
+      else:
+        # Item both in old and new lists -> Nothing to do
+        newTable.del(uid)
+    # newTable now contains items not previously in the old list
+
+    if len(oldIndicesToRemove) > 0:
+      var removedItems = 0
+      for idx in oldIndicesToRemove:
+        let updatedIdx = idx - removedItems
+        self.removeCollectibleItem(updatedIdx)
+        removedItems += 1
```

Comment:
Why are you counting the `removedItems`?

Reply:
`oldIndicesToRemove` holds the indices *of the original list* in sequential ascending order. Every time we delete an item from the list, we should subtract 1 from the remaining indices in  `oldIndicesToRemove`. Instead of doing that I just keep track of how many items I've removed.